File: src/BlazorDB/content/blazorDBInterop.js
Blazor.registerFunction('BlazorDB.blazorDBInterop.SetItem', function (key, value, session) {
    if (session) {
        sessionStorage.setItem(key, value);
    } else {
        localStorage.setItem(key, value);
    }
    return true;
});

Blazor.registerFunction('BlazorDB.blazorDBInterop.GetItem', function (key, session) {
    if (session) {
        return sessionStorage.getItem(key);
    } else {
        return localStorage.getItem(key);
    }
});

Blazor.registerFunction('BlazorDB.blazorDBInterop.RemoveItem', function (key, session) {
    if (session) {
        sessionStorage.removeItem(key);
    } else {
        localStorage.removeItem(key);
    }
    return true;
});


Blazor.registerFunction('BlazorDB.blazorDBInterop.Clear', function (session) {
    if (session) {
        sessionStorage.clear();
    } else {
        localStorage.clear();
    }
    return true;
});

File: src/BlazorDB/Storage/blazorDBInterop.cs
using Microsoft.JSInterop;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace BlazorDB.Storage
{
    internal class BlazorDBInterop : IBlazorDBInterop
    {
        private readonly IJSRuntime _jsRuntime;

        public BlazorDBInterop(IJSRuntime jsRuntime)
        {
            _jsRuntime = jsRuntime;
        }
        public Task<bool> SetItem(string key, string value, bool session)
        {
            return _jsRuntime.InvokeAsync<bool>("blazorDBInterop.setItem", key, value, session);
        }

        public Task<string> GetItem(string key, bool session)
        {
            return _jsRuntime.InvokeAsync<string>("blazorDBInterop.getItem", key, session);
        }

        public Task<bool> RemoveItem(string key, bool session)
        {
            return _jsRuntime.InvokeAsync<bool>("blazorDBInterop.removeItem", key, session);
        }

        public Task<bool> Clear(bool session)
        {
            return _jsRuntime.InvokeAsync<bool>("blazorDBInterop.clear", session);
        }
        public Task<bool> Log(params object[] list)
        {
            List<object> _list = new List<object>(list); //This line is needed see: https://github.com/aspnet/Blazor/issues/740
            return _jsRuntime.InvokeAsync<bool>("blazorDBInterop.logs");
        }
    }
}
